crates/router/src/core/payouts/helpers.rs

```diff
@@ -433,121 +440,246 @@ pub async fn get_or_create_customer_details(
-    };
-
-    let connector_data = api::PayoutConnectorData::get_connector_by_name(
-        &state.conf.connectors,
-        &connector_name,
-        api::GetToken::Connector,
+    // 4. Route connector
+    route_connector_v1(
+        state,
+        merchant_account,
+        &payout_data.business_profile,
+        key_store,
+        &TransactionData::<()>::Payout(payout_data),
+        routing_data,
+        eligible_connectors,
     )
-    .change_context(errors::ApiErrorResponse::InternalServerError)
-    .attach_printable("Routing algorithm gave invalid connector")?;
-
-    routing_data.routed_through = Some(connector_name);
-
-    Ok(api::PayoutConnectorCallType::Single(connector_data))
+    .await
 }
 
 pub async fn get_default_payout_connector(
     _state: &AppState,
     request_connector: Option<serde_json::Value>,
-) -> CustomResult<api::PayoutConnectorChoice, errors::ApiErrorResponse> {
+) -> CustomResult<api::ConnectorChoice, errors::ApiErrorResponse> {
     Ok(request_connector.map_or(
-        api::PayoutConnectorChoice::Decide,
-        api::PayoutConnectorChoice::StraightThrough,
+        api::ConnectorChoice::Decide,
+        api::ConnectorChoice::StraightThrough,
     ))
 }
 
 pub fn should_call_payout_connector_create_customer<'a>(
     state: &AppState,
-    connector: &api::PayoutConnectorData,
+    connector: &api::ConnectorData,
     customer: &'a Option<domain::Customer>,
     connector_label: &str,
 ) -> (bool, Option<&'a str>) {
     // Check if create customer is required for the connector
-    let connector_needs_customer = state
-        .conf
-        .connector_customer
-        .payout_connector_list
-        .contains(&connector.connector_name);
-
-    if connector_needs_customer {
-        let connector_customer_details = customer.as_ref().and_then(|customer| {
-            get_connector_customer_details_if_present(customer, connector_label)
-        });
-        let should_call_connector = connector_customer_details.is_none();
-        (should_call_connector, connector_customer_details)
-    } else {
-        (false, None)
+    match PayoutConnectors::try_from(connector.connector_name) {
+        Ok(connector) => {
+            let connector_needs_customer = state
+                .conf
+                .connector_customer
+                .payout_connector_list
+                .contains(&connector);
+
+            if connector_needs_customer {
+                let connector_customer_details = customer.as_ref().and_then(|customer| {
+                    get_connector_customer_details_if_present(customer, connector_label)
+                });
+                let should_call_connector = connector_customer_details.is_none();
+                (should_call_connector, connector_customer_details)
+            } else {
+                (false, None)
+            }
+        }
+        _ => (false, None),
     }
 }
 
+pub async fn get_gsm_record(
```

Comment:
Can we reuse these functions from payments by moving them to utils / helpers, instead of having redundant code if there's not different logic?

Reply:
Since, different metrics are being used for payments and payouts, the functions are kept distinct